### PR TITLE
331 fix bug adding vis layers to empty layers

### PR DESCRIPTION
**Pull request by longshuicy**
- Default empty layers to a list
- Fix logic to add layers to an empty layer object
- Add new endpoint to get specific visualization GET `/project/api/projects/{projectId}/visualizations/{visualizationId}`
- Cosmetic: move the common field date to parent class "projectResource" that ensures all resource will have a date field

TEST:
- Post a new project: POST `/project/api/projects` 
```
{
  "name": "Example Project",
  "description": "This is a description of the example project.",
  "region": "Joplin"
}
```

- Create a empty visualization POST `/project/api/projects/{projectId}/visualizations`
```
[
    {
    "name": "INCORE Maps",
    "description": "Layers of incore maps",
    "type": "MAP",
    "boundingBox": [
        -90.07376669874641,
        35.03298062856903,
        -89.71464767735003,
        35.207753220358086
    ]
}]
```

- Add layers: POST `/project/api/projects/{projectId}/visualizations/{visualizationId}/layers`
```
[
    {
        "workspace": "incore",
        "layerId": "6375502f3a28a17d261fd682",
        "styleName": "incore:point"
    },
    {
        "workspace": "incore",
        "layerId": "6375502f3a28a17d261fd682",
        "styleName": "incore:point"
    }
]
```

### CODE REVIEW

**Comment by ywkim312** [comment]
github action failed

**Comment by ywkim312** [comment]
In above description for the TEST, the very first step

Post a new project: POST /project/api/projects/{projectId}/

Should this be /project/api/projects? instead of the above?

**Review by ywkim312 — APPROVED** [review]
I followed all the steps for testing and they were all successful.

**Comment by Rashmil-1999** [comment]
Could you also add changelog entry please?

File: server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/models/DatasetResource.java [hunk 1]
```diff
@@ -4,7 +4,6 @@
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
```
Comment (navarroc) [inline server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/models/DatasetResource.java:4]: Can you remove the unused imports?

File: server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/models/VisualizationResource.java [hunk 2]
```diff
@@ -5,7 +5,7 @@
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
```
Comment (navarroc) [inline server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/models/VisualizationResource.java:5]: Can you remove these unused imports too? Thanks.

**Review by Rashmil-1999 — APPROVED** [review]

**Pull request merged**